website/components/LandingPage/OpenSource/index.js

```diff
@@ -55,62 +56,74 @@ function cacheSet(key, data) {
-      if (properties) data = pick(data, properties);
-      cacheSet(key, data);
-      return data;
-    }, console.error);
+  return fallback().then(data => {
+    if (properties) data = pick(data, properties);
+    cacheSet(key, data);
+    return data;
+  }, console.error);
 }
 
-function useStats() {
-  const [stars, setStars] = useState(null);
-  const [forks, setForks] = useState(null);
-  const [downloads, setDownloads] = useState(null);
-
+function getNPMDownloads(from, to) {
   function dateRanges(from, to) {
     from = new Date(from);
     to = new Date(to);
-    const length = to.getFullYear() - from.getFullYear();
+
+    const length =
+      (to.getFullYear() - from.getFullYear()) * 12 +
+      (to.getMonth() - from.getMonth());
+
     const dates = Array.from({ length })
-      .map((_, i) => from.getFullYear() + i + '-01-01')
+      .map(() =>
+        new Date(from.setMonth(from.getMonth() + 1)).toISOString().slice(0, 10)
+      )
       .concat([to.toISOString().slice(0, 10)]);
+
     return dates
       .reduce((acc, curr, i, arr) => acc.concat(curr + ':' + arr[i + 1]), [])
       .slice(0, -1);
   }
 
+  const dates = dateRanges(from, to);
+
+  return Promise.all(
+    dates.map(range =>
+      fetch(`https://api.npmjs.org/downloads/point/${range}/uniforms`)
+        .then(response => response.json())
+        .then(({ downloads }) => downloads || 0)
+    )
+  ).then(sums => sums.reduce((acc, curr) => acc + curr));
+}
+
+function useStats() {
+  const [stars, setStars] = useState(null);
+  const [forks, setForks] = useState(null);
+  const [downloads, setDownloads] = useState(null);
+
   useEffect(() => {
-    cachedFetch('github', 'https://api.github.com/repos/vazco/uniforms', [
-      'stargazers_count',
-      'forks_count'
-    ]).then(({ forks_count: forks, stargazers_count: stars }) => {
+    cached(
+      'github',
+      () =>
+        fetch('https://api.github.com/repos/vazco/uniforms').then(response =>
+          response.json()
+        ),
+      ['stargazers_count', 'forks_count']
+    ).then(({ forks_count: forks, stargazers_count: stars }) => {
       setForks(forks.toLocaleString('en-US'));
       setStars(stars.toLocaleString('en-US'));
     });
   }, [stars, forks]);
 
   useEffect(() => {
-    const from = '2014-01-01';
+    const from = '2016-01-01';
     const today = new Date().toISOString().slice(0, 10);
-    const dates = dateRanges(from, today);
-
-    Promise.all(
-      dates.map(range =>
-        cachedFetch(
-          `npm-${range}`,
-          `https://api.npmjs.org/downloads/point/${range}/uniforms`,
-          ['downloads']
-        ).then(({ downloads }) => downloads || 0)
-      )
-    )
-      .then(sums => sums.reduce((acc, curr) => acc + curr))
-      .then(downloads => setDownloads(downloads.toLocaleString('en-US')));
+
+    cached(`npm-${from}--${today}`, () => getNPMDownloads(from, today)).then(
+      downloads => setDownloads(downloads.toLocaleString('en-US'))
+    );
```

Comment:
Maybe it's too much to do it now, but we could cache them per-month for longer and then just update the latest data.